debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java

```diff
@@ -344,12 +354,41 @@ public static ConfigDef configDef() {
     private final SourceTimestampMode sourceTimestampMode;
     private final boolean readOnlyDatabaseConnection;
     private final int maxTransactionsPerIteration;
+    private final boolean multiPartitionMode;
 
     public SqlServerConnectorConfig(Configuration config) {
         super(SqlServerConnector.class, config, config.getString(SERVER_NAME), new SystemTablesPredicate(), x -> x.schema() + "." + x.table(), true,
                 ColumnFilterMode.SCHEMA);
 
-        this.databaseName = config.getString(DATABASE_NAME);
+        boolean hasDatabaseName = config.hasKey(DATABASE_NAME.name());
+        boolean hasDatabaseNames = config.hasKey(DATABASE_NAMES.name());
+
+        if (hasDatabaseName && hasDatabaseNames) {
+            throw new IllegalArgumentException("Only one of the '" + DATABASE_NAME + "' and '" + DATABASE_NAMES
+                    + "' options can be specified at a time");
+        }
```

Comment:
Let's push this into a field validation rule in `DATABASE_NAMES` (e.g. similar to include/exclude list options which also are mutually exclusive). That way, configuration validation in Kafka Connect will flag that correctly.

Reply:
Done. Also added a test.